WordPress/gutenberg
# [RNMobile] Handle undefined block case in block actions menu

`gutenberg-mobile` PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/3389





## Description


### Problem
I was testing the undo/redo functionality with the debugger enabled and I realised that when undoing a block creation an exception is raised in the blocks actions menu component.

**Exception:**
`TypeError: Cannot read property 'title' of undefined`
[packages/block-editor/src/components/block-mobile-toolbar/block-actions-menu.native.js#L275](packages/block-editor/src/components/block-mobile-toolbar/block-actions-menu.native.js#L275)

This exception is not leading to a crash fortunately but it could produce undesired side effects as well as make the debugging experience a bit hard.

I checked further and the cause is that the block can be `undefined` right after the undo action is executed. Some of the calculations made in the [`withSelect` block](https://github.com/WordPress/gutenberg/blob/aa272f8432ca055961005a62c200a856e8017cb7/packages/block-editor/src/components/block-mobile-toolbar/block-actions-menu.native.js#L271-L297) are not checking this case so they produce the TypeError exceptions.

### Solution

I checked the logic of that part and I added conditions for handling the undefined block case.

## How has this been tested?



NOTE: The exception can only be triggered enabling the debugging mode via the RN developer menu.

1. Create a post
2. Open the RN developer menu and tap on Debug button
3. [On the debugger window] Go to the `Sources` tab and enable `Pause on exception` (see attached screenshot)
<img width="715" alt="Screenshot 2021-04-16 at 17 59 29" src="https://user-images.githubusercontent.com/14905380/115051847-8c13c900-9edd-11eb-9351-db8e45bf9b99.png">
4. Tap on the initial paragraph block
5. Add some text
6. Duplicate the block
7. Undo the action
8. Observe that a exception is raised pointing to `block-actions-menu.native.js` file

## Screenshots 

https://user-images.githubusercontent.com/14905380/115052892-c467d700-9ede-11eb-9901-63650249241a.mp4

## Types of changes




Bug fix

## Checklist:
